src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -737,4 +726,19 @@ private boolean isMissingLocation(PoLine poLine, Piece piece) {
       }
     });
   }
+
+  CompletionStage<List<CompletableFuture<Void>>> removeForbiddenEntities(List<PoLine> poLines, Map<String, Map<String, T>> pieces) {
+    Map<String, List<PoLine>> poLinesGroupedByOrderId = poLines.stream().collect(groupingBy(PoLine::getPurchaseOrderId));
+    String query = buildQuery(convertIdsToCqlQuery(new ArrayList<>(poLinesGroupedByOrderId.keySet())), logger);
+    String url = String.format(GET_PURCHASE_ORDERS, Integer.MAX_VALUE, 0, query, lang);
+    return handleGetRequest(url, httpClient, ctx, okapiHeaders, logger)
+      .thenApply(orders -> orders.mapTo(PurchaseOrders.class).getPurchaseOrders().stream().map(order -> protectionHelper.isOperationRestricted(order.getAcqUnitIds(), ProtectedOperationType.UPDATE)
+        .exceptionally(t -> {
+          poLinesGroupedByOrderId.get(order.getId())
+            .forEach(line -> Objects.requireNonNull(pieces.computeIfPresent(line.getId(), (s, stringCheckInPieceMap) -> pieces.remove(line.getId()))).keySet()
+              .forEach(pieceId -> addError(line.getId(), pieceId, USER_HAS_NO_PERMISSIONS.toError())));
+          return null;
+        }))
+        .collect(toList()));
```

Comment:
The result of this is not used anyhow. Please consider just to use `CompletableFuture.allOf` just to return `CompletableFuture<Void>` as a result